.github/workflows/run-cctp-tests.yml

```diff
@@ -0,0 +1,95 @@
+name: PR approved workflow
+
+on:
+  issue_comment:
+    types:
+      - created
+  pull_request_review:
+    types:
+      - submitted
+  push:
+    branches: ["master"]
+  workflow_dispatch:
+  merge_group:
+
+# https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#concurrency
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}-cctp
+  cancel-in-progress: true
+
+env:
+  NEXT_PUBLIC_INFURA_KEY: ${{ secrets.NEXT_PUBLIC_INFURA_KEY }}
+  NEXT_PUBLIC_WALLETCONNECT_PROJECT_ID: ${{ secrets.NEXT_PUBLIC_WALLETCONNECT_PROJECT_ID }}
+  THE_GRAPH_NETWORK_API_KEY: ${{ secrets.THE_GRAPH_NETWORK_API_KEY }}
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  IS_HOTFIX: ${{ contains(github.event.pull_request.title, 'hotfix') }}
+
+jobs:
+  check-is-hotfix:
+    name: Check if PR is hotfix
+    runs-on: ubuntu-latest
+    outputs:
+      is_hotfix: ${{ steps.check-is-hotfix.outputs.is_hotfix }}
+    env:  
+      IS_HOTFIX: ${{ contains(github.event.pull_request.title, 'hotfix') }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./.github/actions/check-is-hotfix
+        id: check-is-hotfix
+
+  check-files:
+    name: Check files
+    runs-on: ubuntu-latest
+    outputs:
+      run_tests: ${{ steps.check-files.outputs.run_tests }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./.github/actions/check-files
+        id: check-files
+
+  build:
+    name: "Build"
+    runs-on: ubuntu-latest
+    needs: [check-files, check-is-hotfix]
+    if: needs.check-files.outputs.run_tests == 'true' && needs.check-is-hotfix.outputs.is_hotfix == 'false'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install node_modules
+        uses: OffchainLabs/actions/node-modules/install@main
+
+      - name: Build
+        run: yarn build
+        env:
+          NEXT_PUBLIC_IS_E2E_TEST: true
+          NEXT_PUBLIC_INFURA_KEY: ${{ secrets.NEXT_PUBLIC_INFURA_KEY }}
+          NEXT_PUBLIC_WALLETCONNECT_PROJECT_ID: ${{ secrets.NEXT_PUBLIC_WALLETCONNECT_PROJECT_ID }}
+          THE_GRAPH_NETWORK_API_KEY: ${{ secrets.THE_GRAPH_NETWORK_API_KEY }}
+
+      - name: Cache build artifacts
+        uses: ./.github/actions/build-artifacts/cache
+
+  cctp-e2e-tests:
+    name: "CCTP E2E Tests"
+    needs: [build, check-is-hotfix, check-files]
+    # Only run CCTP tests when the PR was just approved
+    if: needs.check-is-hotfix.outputs.is_hotfix == 'false' && needs.check-files.outputs.run_tests == 'true' && github.event_name == 'pull_request_review' && github.event.review.state == 'approved'
```

Comment:
doesn't look like `issue_comment` would trigger this workflow, so should we remove it from L4?
and if so, we can also remove `github.event_name == 'pull_request_review'` from here because this whole workflow will only be triggered by this event

and we want to allow `workflow_dispatch` to trigger this?